api/client/service/update.go

```diff
@@ -189,65 +198,132 @@ func updateService(flags *pflag.FlagSet, spec *swarm.ServiceSpec) error {
 	return nil
 }
 
-func updateLabels(flags *pflag.FlagSet, field *map[string]string) {
-	if !flags.Changed(flagLabel) {
-		return
+func anyChanged(flags *pflag.FlagSet, fields ...string) bool {
+	for _, flag := range fields {
+		if flags.Changed(flag) {
+			return true
+		}
 	}
+	return false
+}
 
-	values := flags.Lookup(flagLabel).Value.(*opts.ListOpts).GetAll()
+func updatePlacement(flags *pflag.FlagSet, placement *swarm.Placement) {
+	field, _ := flags.GetStringSlice(flagConstraintAdd)
+	constraints := &placement.Constraints
+	placement.Constraints = append(placement.Constraints, field...)
 
-	localLabels := map[string]string{}
-	for key, value := range runconfigopts.ConvertKVStringsToMap(values) {
-		localLabels[key] = value
+	toRemove := buildToRemoveSet(flags, flagConstraintRemove)
+	for i, constraint := range placement.Constraints {
+		if _, exists := toRemove[constraint]; exists {
+			*constraints = append((*constraints)[:i], (*constraints)[i+1:]...)
```

Comment:
This is the "delete" recipe from https://github.com/golang/go/wiki/SliceTricks

I don't think it's overwriting part of the slice, it's assigning a new slice to the old variable from two parts of the original slice.

`slice.Delete(x)` would be great...